### PR TITLE
Simple Animal Bots [WIP]

### PR DESCRIPTION
**Pull request by AnturK**
Am i insane ? Possibly.

### CODE REVIEW

**Comment by WJohn** [comment]
Oh my god.

File: code/modules/mob/living/carbon/human/species.dm [hunk 1]
```diff
@@ -708,9 +708,14 @@
 			H.client.screen += global_hud.vimpaired
 		if(H.eye_blurry)			H.client.screen += global_hud.blurry
 		if(H.druggy)
-			H.client.screen += global_hud.druggy
+			var/list/drug_colors = list()
```
Comment (RemieRichards) [inline code/modules/mob/living/carbon/human/species.dm]: this is probably not for this PR

File: code/modules/mob/living/life.dm [hunk 2]
```diff
@@ -202,9 +202,15 @@
 					client.screen += global_hud.blurry
 
 				if (druggy)
-					client.screen += global_hud.druggy
+					//client.screen += global_hud.druggy
+					var/list/drug_colors = list()
```
Comment (RemieRichards) [inline code/modules/mob/living/life.dm]: or this

File: code/datums/supplypacks.dm [hunk 3]
```diff
@@ -1026,7 +1026,7 @@ var/list/all_supply_groups = list(supply_emergency,supply_security,supply_engine
 
 /datum/supply_packs/misc/mule
 	name = "MULEbot Crate"
-	contains = list(/obj/machinery/bot/mulebot)
+	contains = list(/mob/living/simple_animal/bot/mulebot)
```
Comment (Anonus) [inline code/datums/supplypacks.dm]: Unless someone has changed the supply shuttle, these need to do what the animal crates do and put the animal in after it arrives on station.

Comment (AnturK) [inline code/datums/supplypacks.dm]: Well emergency crate worked for me. Didn't try the mulebot one. Will check in a moment.

File: code/modules/mob/living/simple_animal/bot/mulebot.dm [hunk 4]
```diff
@@ -1,866 +1,810 @@
...
+		var/turf/T = loc
+		var/turf/newT = get_step(T,dirn)
+		if(load.CanPass(load,newT)) //Can't get off onto anything that wouldn't let you pass normally
+			step(load, dirn)
+	if(passenger)
+		if(passenger.client)
+			passenger.client.eye = passenger.client.mob
+			passenger.client.perspective = MOB_PERSPECTIVE
+		passenger = null
+	load = null
```
Comment (phil235) [inline code/modules/mob/living/simple_animal/bot/mulebot.dm]: So you can't get on top of the mulebot anymore, you're inside of it? It's not consistent with how crates are loaded.

**Comment by GunHog** [comment]
I still do not see what the point of doing this is. What is the benefit?

File: code/modules/mob/living/simple_animal/bot/mulebot.dm [hunk 5]
```diff
@@ -1,866 +1,810 @@
...
+		if(!can_step)
+			return 0
+	*/
+	if(M.client)
+		M.client.perspective = EYE_PERSPECTIVE
+		M.client.eye = src
+	passenger = M
+	load = M
+	M.loc = src
+	return 1
```
Comment (phil235) [inline code/modules/mob/living/simple_animal/bot/mulebot.dm]: So you can't get on top of the mulebot anymore, you're inside of it? It's not consistent with how crates are loaded.

Comment (AnturK) [inline code/modules/mob/living/simple_animal/bot/mulebot.dm]: Crates work the same now. I guess i could port buckling to mobs but that seems overkill.

Comment (phil235) [inline code/modules/mob/living/simple_animal/bot/mulebot.dm]: Having a mob inside the mulebot means they can drop items inside the bot. See https://github.com/tgstation/-tg-station/pull/11104/files#diff-c9ce05dfd7137c9de66315892904a605L484

Comment (AnturK) [inline code/modules/mob/living/simple_animal/bot/mulebot.dm]: Good point. Will do something about it.

**Comment by Niknakflak** [comment]
It's benifical because now people can play as bots

File: code/modules/mob/living/simple_animal/bot/mulebot.dm (hunk 5 above)
Comment (phil235) [inline code/modules/mob/living/simple_animal/bot/mulebot.dm]: If you're up for it, you can move the buckle code from /obj to the /atom/movable level, so you can basically keep the mob buckling to mulebot code. (Plus I need it to make slime glomping use buckling, which will simplify plenty of things)

**Comment by Xhuis** [comment]
emagging beepsky puts a ghost into it when

**Comment by Fox-McCloud** [comment]
Aside from the "can play as bots" thing--yeah, what's the benefit to this?

**Comment by PKPenguin321** [comment]
it standardizes bots, hopefully avoiding the 10,000 issues that pop up with them every other week
also why not, the code's all right here and from a player perspective it shouldn't change anything

**Comment by Fox-McCloud** [comment]
But....bots are already standardized?

**Comment by PKPenguin321** [comment]
yeah but they could be *more* standardized

**Comment by Fox-McCloud** [comment]
they're all bots...and all the bots are subtypes of that. I dunno how more standardized you could get. Again, I see no point do this unless it has performance+behavioral advantages to it.

File: code/controllers/subsystem/bots.dm [hunk 6]
```diff
@@ -1,22 +0,0 @@
...
-	NEW_SS_GLOBAL(SSbot)
-
-/datum/subsystem/bots/stat_entry(msg)
-	..("P:[processing.len]")
-
-/datum/subsystem/bots/fire()
-	var/seconds = wait * 0.1
-	for(var/thing in processing)
-		if(thing && !thing:gc_destroyed)
-			spawn(-1)
```
Comment (Fox-McCloud) [inline code/controllers/subsystem/bots.dm]: this is important--bot code utilizes sleeps in it, thus the reason why each of these are spawned off like this---if you move bots over to be mobs and retain those sleeps, it's going to cause issues.

**Comment by Jordie0608** [comment]
I remember Pete once told me this would be a great idea but I can't remember why.

**Comment by AnturK** [comment]
Standardized Stat,Damage,Interaction with other mobs,hopefully movement in the future. No need to sprinkle type checks everywhere for them. They share more functionality with mobs than machines.

File: code/modules/mob/living/simple_animal/bot/mulebot.dm (hunk 5 above)
Comment (AnturK) [inline code/modules/mob/living/simple_animal/bot/mulebot.dm]: After looking at it, yeah that might be cleanest solution. I'll see how that works out.

**Comment by Incomptinence** [comment]
If you are standardizing bots you might want to look at the mining bot. The bot that isn't a bot and its totally different interface scheme (ta ergo). Also some bug is screwing with their repairs.

**Comment by WJohn** [comment]
What's the status on this?

**Comment by AnturK** [comment]
Goddamit Metastation

**Comment by Midaychi** [comment]
You could probably implement your own version of this, if you made bots simple animals. https://github.com/d3athrow/vgstation13/pull/5000

**Comment by Razharas** [comment]
Hows the progress?
@AnturK

**Comment by Jordie0608** [comment]
Testing round:
Turrets shoot at bots

**Comment by Anonus** [comment]
Players being assholes is a feature.

**Comment by Bawhoppen** [comment]
this is best feature pls fix and merge soon

**Comment by AnturK** [comment]
I give up

**Comment by PKPenguin321** [comment]
reeeeeeeeeeeeeee

**Comment by WJohn** [comment]
RIP the dream. Hopefully someone will pick this up.

**Comment by KorPhaeron** [comment]
@AnturK 

Dont let the dream die

**Comment by Bawhoppen** [comment]
arguably most important thing to get added

**Pull request closed without merging**